src/content/index.js

```diff
@@ -1,72 +1,141 @@
+  stopScrollingOnHover: false
+};
+
+let autoScrolling = Object.assign({}, defaultValues, {
+  scrollingAction: function() {
+    if (this.stopScrollingOnHover && this.currentlyHovering) {
+      return;
+    }
     this.y = this.y + this.scrollingStep;
     this.scrollingElement.scroll(this.x, this.y);
-    this.tid = setTimeout(() => {
-      this.start();
-    }, 100 - this.scrollingSpeed);
   },
-  stop: function () {
-    clearTimeout(this.tid);
-    this.tid = -1;
+  start: function() {
+    this.scrollingElement = getScrollingElement();
+    unregisterMouseListeners();
+    registerMouseListeners();
+    this.intervalId = window.setInterval(
+      this.scrollingAction.bind(this),
+      100 - this.scrollingSpeed
+    );
   },
+  stop: function() {
+    unregisterMouseListeners();
+    window.clearInterval(this.intervalId);
+  }
+});
+
+// Deconstruct only the fields which are synchronized.
+const updateFromSync = () => {
+  const {
+    scrollingStep,
+    scrollingSpeed,
+    stopScrollingByClick,
+    stopScrollingOnHover
+  } = defaultValues;
+
+  browser.storage.sync
+    .get({
+      scrollingStep,
+      scrollingSpeed,
+      stopScrollingByClick,
+      stopScrollingOnHover
+    })
+    .then(data => {
+      Object.assign(autoScrolling, data);
+    })
+    .catch(onError);
 };
 
-const getScrollingElement = () => {
-  return document.scrollingElement ?
-    document.scrollingElement : document.documentElement;
+updateFromSync();
+
+// DOM Events
+const mouseoverListeners = ev => {
+  const target = ev.target;
+  if (!autoScrolling.stopScrollingOnHover) {
+    return;
+  }
+  if (target == document.body) {
+    autoScrolling.currentlyHovering = false;
+  } else {
+    let targetRect = target.getBoundingClientRect();
+    // Check if the mouse is overlapping with the element's dimensions.
+    if (
+      targetRect.width != document.body.clientWidth &&
+      (targetRect.right > ev.pageX || targetRect.top > ev.pageY)
+    ) {
+      autoScrolling.currentlyHovering = true;
+    } else {
+      autoScrolling.currentlyHovering = false;
+    }
+  }
 };
 
-const defaultStopScrollingByClick = true;
-browser.storage.sync.get({
-  stopScrollingByClick: defaultStopScrollingByClick
-}).then((data) => {
-  const { stopScrollingByClick } = data;
-  autoScrolling.stopScrollingByClick = stopScrollingByClick;
-}).catch(onError);
+const mouseoutListeners = () => {
+  autoScrolling.currentlyHovering = false;
+};
 
-const defaultScrollingSpeed = 50;
-browser.storage.sync.get({
-  scrollingSpeed: defaultScrollingSpeed
-}).then((data) => {
-  const { scrollingSpeed } = data;
-  autoScrolling.scrollingSpeed= scrollingSpeed;
-}).catch(onError);
+const clickListeners = () => {
+  if (autoScrolling.stopScrollingByClick == true) {
+    browser.runtime
+      .sendMessage({
+        isScrolling: false
+      })
+      .then(() => {
+        autoScrolling.stop();
+      })
+      .catch(onError);
+  }
+};
 
+const registerMouseListeners = () => {
```

Comment:
@jpsouzasilva I think `addMouseEventListeners` is better. 
Similarly, `removeMouse...` is better than `unregister...`.